### ISSUE TITLE
The documentation for errorObserver doesn't appear to be correct

### ISSUE THREAD

**Issue by simonckenyon** [marcglasberg/async_redux#82]
If I define an errorObserver as per the documentation I get a compilation error:

The argument type 'bool Function(Object, ReduxAction<dynamic>, Store<dynamic>, Object, int)' can't be assigned to the parameter type 'ErrorObserver<dynamic>'.

**Comment by marcglasberg** [comment]
Yes, the docs were outdated. Thanks for noticing. It should be this:

```dart
var store = Store<AppState>(
  initialState: AppState.initialState(),
  errorObserver: MyErrorObserver<AppState>(),
);

class MyErrorObserver<St> implements ErrorObserver<St> {
  @override
  bool observe(Object error, StackTrace stackTrace, ReduxAction<St> action, Store store) {
    print("Error thrown during $action: $error");
    return true;
  }
}
```

**Comment by marcglasberg** [comment]
Fix version:  2.12.3
